scripts/download.sh

```diff
@@ -25,6 +46,21 @@ if [ ! -f /var/spool/ctlstore/ldb.db ]; then
     COMPRESSED="true"
   fi
 
+  if [ -z $remote_checksum ]; then
+    echo "Remote checksum is null, skipping checksum validation"
+  else
+    local_checksum=$(shasum -a 256 snapshot.db | cut -f1 -d\ | xxd -r -p | base64)
```

Comment:
Not blocking/idea for further optimization: how long does the sha256 of the ldb file take?

We don't need a cryptographically secure hash here, and might be better served by something that's fast (SHA-1 would likely be ~2x faster, and something like FNV even faster).

Reply:
```
│ Checksum matches                                                                                                                                                                                             │
│ Local checksum calculation took 13 seconds                                                                                                                                                                   │
│ ldb.db ready in 68 seconds
```
Sha1 only took 13 sec in prod-usw2, not much overhead at all, I am pretty satisfactory with this result, such a relive 😌

Reply:
I was just pondering about **how** we could migrate to [FNV](https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function) (as Andrew suggested), if we want to in the future -- I think it's ok to leave the Metadata name as `checksum` for now and assume that means `sha1`.  If we add `FNV` support in the future we'd need to distinguish the checksum header between the `sha1` and `FNV` headers, could be done by calling the `FNV` header `checksum_fnv` or something similar.

Reply:
Nice, yeah something good to have in our back pocket. For posterity: https://github.com/cespare/xxhash might be the best result (fastest) - and it could bring us down to <1s for the checksum. We use it in integrations-consumer as the hash function for our consistent hashing scheme.

Reply:
TIL. <1s would be pretty decent, but yea I am also leaning towards leaving it for future optimization when we have free cycle unless you are strongly opinionated on getting it done in this iteration. Created ticket for this: https://segment.atlassian.net/browse/CONFIG-332

Reply:
I think getting the big win out is more important than spinning another iteration *right now*.

Reply:
BTW, `sha256` took about ~32 seconds, so `sha1` is definitely what we want for now.